### PR TITLE
[WIP/TEST] Build kebabRules map

### PR DESCRIPTION
**Pull request by Markionium**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Include a change request file using `$ yarn change`

#### Description of changes

(give an overview)

#### Focus areas to test

(optional)

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/10422)

### CODE REVIEW

**Comment by KevinTCoughlin** [comment]
@Markionium let me know when to loop in folks for review. This is most likely something @dzearing @jasongore should look at if you wish to pursue it (building the map once at runtime).

**Comment by JasonGore** [comment]
That's interesting, I think the regression analysis is potentially showing the static init hit on perf. I wonder if there's any value on lazy building this?

File: packages/merge-styles/src/transforms/kebabRules.ts [hunk 1]
```diff
@@ -1,124 +1,16 @@
...
+  let key: string;
+  for (key in elementStyles) {
+    if (key) {
+      kebabMap[key] = key.replace(/([A-Z])/g, '-$1').toLowerCase();
+    }
+  }
+  return kebabMap;
+}
+
+const rules: { [key: string]: string } = makeKebabMap();
```
Comment (dzearing) [inline packages/merge-styles/src/transforms/kebabRules.ts]: I don't think this is necessary. If you remove this code and leave it as a blank object, We made a change to cache things as needed. I'm going to try and tweak your PR.

Comment (dzearing) [reply packages/merge-styles/src/transforms/kebabRules.ts]: I modified the PR to just let it lazy cache. This is good enough :)

**Comment by dzearing** [comment]
LGTM!

**Pull request merged**